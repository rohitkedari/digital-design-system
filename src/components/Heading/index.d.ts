import React from 'react';

export interface HeadingProps {
	color?: string;
	level?: '1' | '2' | '3' | '4' | '5' | '6' | 1 | 2 | 3 | 4 | 5 | 6;
	size?: 'small' | 'medium' | 'large' | 'xlarge' | string;
	textAlign?: TextAlignType;
}
declare const Heading: React.SFC<HeadingProps>;

export { Heading };
