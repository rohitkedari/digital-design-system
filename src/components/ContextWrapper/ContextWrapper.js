/**
 * context wrapper component
 */
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeContext, ResponsiveContextProvider } from 'contexts';
import { deepMerge, GlobalStyle, loadFonts } from 'utils';
import { baseTheme } from 'themes/base';

const ContextWrapper = ({ children, theme }) => {
    const nextTheme = deepMerge(baseTheme, theme);
    loadFonts(nextTheme.global.font.url);
    return (
        <ThemeContext.Provider value={nextTheme}>
            <ResponsiveContextProvider
                breakpoints={nextTheme.global.breakpoints}
            >
                <GlobalStyle />
                {children}
            </ResponsiveContextProvider>
        </ThemeContext.Provider>
    );
};
ContextWrapper.defaultProps = {
    theme: null,
};
ContextWrapper.propTypes = {
    children: PropTypes.node.isRequired,
    theme: PropTypes.shape({}),
};
export { ContextWrapper };
