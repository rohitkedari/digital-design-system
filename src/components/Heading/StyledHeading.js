import styled, { css } from 'styled-components';
import { getColor } from 'utils';

const sizeStyle = (props) => {
	// size is a combination of the level and size properties
	const size = props.size || 'medium';
	const headingTheme = props.theme.heading;
	const levelStyle = headingTheme.level[props.level];
	if (levelStyle) {
		const data = levelStyle[size];
		const styles = [
			css`
				font-size: ${data ? data.size : size};
				line-height: ${data ? data.height : 'normal'};
				max-width: ${data ? data.maxWidth : levelStyle.medium.maxWidth};
				font-weight: ${levelStyle.font.weight || headingTheme.weight};
			`,
		];
		return styles;
	}
	return '';
};

const fontFamily = (props) => {
	const { font } = props.theme.heading.level[props.level];
	if (font && font.family) {
		return css`
			font-family: ${font.family};
		`;
	}
	return props.theme.heading.font
		? css`
				font-family: ${props.theme.heading.font.family};
		  `
		: '';
};

const StyledHeading = styled.h4`
	${(props) => fontFamily(props)}
	${(props) => sizeStyle(props)}
	text-align: ${(props) => props.textAlign};
	color: ${(props) => getColor(props.colorProp, props.theme)};
`;

export { StyledHeading };
