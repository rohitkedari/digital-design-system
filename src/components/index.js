/* GENERATOR: Assemble all components for export */
export * from './Modal';
export * from './Select';
export * from './ContextWrapper';
export * from './Heading';
export * from './Paragraph';
export * from './Icon';
