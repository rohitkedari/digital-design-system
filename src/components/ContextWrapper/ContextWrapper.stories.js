import React, { useContext } from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { ThemeContext, ResponsiveContext } from 'contexts';
import { withA11y } from '@storybook/addon-a11y';
import { ContextWrapper } from './ContextWrapper';

const StyledContent = styled('div')`
	color: ${(props) => props.color || props.theme.global.colors.brand};
`;
storiesOf('components|ContextWrapper', module)
	.addParameters({ component: ContextWrapper })
	.addDecorator(withA11y)
	.add('Default Context', () => {
		const Content = () => {
			const { viewport } = useContext(ResponsiveContext);
			const theme = useContext(ThemeContext);
			return (
				<div>
					{`This is a ${viewport} screen`}
					<StyledContent>Text with brand color</StyledContent>
					<StyledContent color={theme.global.colors.status_ok}>
						Text with success color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_error}>
						Text with error color
					</StyledContent>
					<StyledContent color={theme.global.colors.status_warning}>
						Text with warning color
					</StyledContent>
				</div>
			);
		};
		return (
			<ContextWrapper>
				<Content />
			</ContextWrapper>
		);
	});
