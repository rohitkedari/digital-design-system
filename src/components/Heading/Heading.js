import React from 'react';
import PropTypes from 'prop-types';
import { StyledHeading } from './StyledHeading';

const Heading = ({ level, size, color, textAlign, ...rest }) => {
	return (
		<StyledHeading
			as={`h${level}`}
			size={size}
			colorProp={color}
			level={+level}
			textAlign={textAlign}
			{...rest}
		/>
	);
};

Heading.defaultProps = {
	size: 'medium',
	level: 4,
	color: null,
	textAlign: null,
};

Heading.propTypes = {
	size: PropTypes.oneOf([
		'small',
		'medium',
		'large',
		'xlarge',
		PropTypes.string,
	]),
	level: PropTypes.number,
	color: PropTypes.string,
	textAlign: PropTypes.oneOf(['center', 'right', 'left']),
};

export { Heading };
