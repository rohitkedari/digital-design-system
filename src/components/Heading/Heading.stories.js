import React from 'react';
import { storiesOf } from '@storybook/react';
import { withTests } from '@storybook/addon-jest';
import { Heading } from './Heading';
import results from './__tests__/Heading-test-results.json';

storiesOf('components|Heading', module)
	.addDecorator(
		withTests({
			results,
		}),
	)
	.addParameters({ component: Heading, jest: ['Heading.test.js'] })
	.add('All Default Headings', () => (
		<div>
			<Heading>Default Heading</Heading>
			<Heading level={1}>Heading H1</Heading>
			<Heading level={2}>Heading H2</Heading>
			<Heading level={3}>Heading H3</Heading>
			<Heading level={4}>Heading H4</Heading>
			<Heading level={5}>Heading H5</Heading>
			<Heading level={6}>Heading H6</Heading>
		</div>
	))
	.add('Aligned Headings', () => (
		<div>
			<Heading textAlign="center" level={1}>
				Heading H1 center
			</Heading>
			<Heading textAlign="left" level={2}>
				Heading H2 left
			</Heading>
			<Heading textAlign="right" level={3}>
				Heading H3 right
			</Heading>
		</div>
	))
	.add('Colored Headings', () => (
		<div>
			<Heading color="brand" level={1}>
				Heading H1 brand
			</Heading>
			<Heading color="#95591E" level={2}>
				Heading H2 custom
			</Heading>
			<Heading color="status_ok" level={3}>
				Heading H3 success
			</Heading>
		</div>
	))
	.add('Small Headings', () => (
		<div>
			<Heading size="small">Small Heading</Heading>
			<Heading size="small" level={1}>
				Heading H1
			</Heading>
			<Heading size="small" level={2}>
				Heading H2
			</Heading>
			<Heading size="small" level={3}>
				Heading H3
			</Heading>
			<Heading size="small" level={4}>
				Heading H4
			</Heading>
			<Heading size="small" level={5}>
				Heading H5
			</Heading>
			<Heading size="small" level={6}>
				Heading H6
			</Heading>
		</div>
	))
	.add('Large Headings', () => (
		<div>
			<Heading size="large">Large Heading</Heading>
			<Heading size="large" level={1}>
				Heading H1
			</Heading>
			<Heading size="large" level={2}>
				Heading H2
			</Heading>
			<Heading size="large" level={3}>
				Heading H3
			</Heading>
			<Heading size="large" level={4}>
				Heading H4
			</Heading>
			<Heading size="large" level={5}>
				Heading H5
			</Heading>
			<Heading size="large" level={6}>
				Heading H6
			</Heading>
		</div>
	))
	.add('xLarge Headings', () => (
		<div>
			<Heading size="xlarge">Default Heading</Heading>
			<Heading size="xlarge" level={1}>
				Heading H1
			</Heading>
			<Heading size="xlarge" level={2}>
				Heading H2
			</Heading>
			<Heading size="xlarge" level={3}>
				Heading H3
			</Heading>
			<Heading size="xlarge" level={4}>
				Heading H4
			</Heading>
			<Heading size="xlarge" level={5}>
				Heading H5
			</Heading>
			<Heading size="xlarge" level={6}>
				Heading H6
			</Heading>
		</div>
	));
