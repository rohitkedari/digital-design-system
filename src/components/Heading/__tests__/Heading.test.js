import React from 'react';
import renderer from 'react-test-renderer';
import { ContextWrapper } from 'components/ContextWrapper';
import { Heading } from '../Heading';

describe('Heading', () => {
	test('with default renders correctly', () => {
		const tree = renderer
			.create(
				<ContextWrapper>
					<Heading>Test Heading</Heading>
				</ContextWrapper>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
