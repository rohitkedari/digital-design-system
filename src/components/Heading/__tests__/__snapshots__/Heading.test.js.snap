// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading with default renders correctly 1`] = `
<h4
  className="sc-AxjAm bJtAKX"
  size="medium"
>
  Test Heading
</h4>
`;
